__tests__/integration/address-info.test.js

```diff
@@ -0,0 +1,145 @@
+    const response = await TestUtils.request
+      .get("/wallet/address-info")
+      .query({address: await wallet1.getAddressAt(1)})
+      .set({"x-wallet-id": wallet1.walletId});
+
+    expect(response.status).toBe(200);
+
+    const results = response.body;
+    expect(results.success).toBeTruthy();
+    expect(results.token).toBe(HATHOR_TOKEN_ID)
+    expect(results.index).toBe(1);
+    expect(results.total_amount_received).toBe(address1balance)
+    expect(results.total_amount_sent).toBe(0)
+    expect(results.total_amount_available).toBe(address1balance)
+    expect(results.total_amount_locked).toBe(0)
+    done();
+  });
+
+  it("should return results for an address with send/receive transactions", async done => {
+    const response = await TestUtils.request
+      .get("/wallet/address-info")
+      .query({address: await wallet2.getAddressAt(0)})
+      .set({"x-wallet-id": wallet2.walletId});
+
+    expect(response.status).toBe(200);
+
+    const results = response.body;
+    expect(results.success).toBeTruthy();
+    expect(results.token).toBe(HATHOR_TOKEN_ID)
+    expect(results.index).toBe(0);
+    expect(results.total_amount_received).toBe(15) // 10 from genesis, 5 from token creation change
+    expect(results.total_amount_sent).toBe(10) // token creation tx
+    expect(results.total_amount_available).toBe(5) // change
+    expect(results.total_amount_locked).toBe(0)
+    done();
+  });
+
+  it("should return results for custom token for an address (empty)", async done => {
+    const response = await TestUtils.request
+      .get("/wallet/address-info")
+      .query({
+        address: await wallet2.getAddressAt(0),
+        token: customTokenHash,
+      })
+      .set({"x-wallet-id": wallet2.walletId});
+
+    expect(response.status).toBe(200);
+
+    const results = response.body;
+    expect(results.success).toBeTruthy();
+    expect(results.token).toBe(customTokenHash);
+    expect(results.index).toBe(0);
+    expect(results.total_amount_received).toBe(0)
+    expect(results.total_amount_sent).toBe(0)
+    expect(results.total_amount_available).toBe(0)
+    expect(results.total_amount_locked).toBe(0)
+    done();
+  });
+
+  it("should return results for custom token on an address with a single transaction", async done => {
+    const response = await TestUtils.request
+      .get("/wallet/address-info")
+      .query({
+        address: await wallet2.getAddressAt(1),
+        token: customTokenHash,
+      })
+      .set({"x-wallet-id": wallet2.walletId});
+
+    expect(response.status).toBe(200);
+
+    const results = response.body;
+    expect(results.success).toBeTruthy();
+    expect(results.token).toBe(customTokenHash)
+    expect(results.index).toBe(1);
+    expect(results.total_amount_received).toBe(500)
+    expect(results.total_amount_sent).toBe(0)
+    expect(results.total_amount_available).toBe(500)
+    expect(results.total_amount_locked).toBe(0)
+    done();
+  });
+
```

Comment:
Maybe add some locked amount in one of the tests as well.

Reply:
A new test was added to validate the locked amount on the miner rewards address.